### PR TITLE
node: Add ability to promisify setTimeout and setImmediate

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/16860#issuecomment-325312379
Documentation: https://nodejs.org/api/timers.html#timers_settimeout_callback_delay_args

CC @jrobeson

### CODE REVIEW

**Comment by wwwy3y3** [comment]
seems like `__promisify__` is a workaround. LGTM 👍

File: types/node/index.d.ts [hunk 1]
```diff
@@ -81,10 +81,16 @@ declare var __filename: string;
 declare var __dirname: string;
 
 declare function setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timer;
+declare namespace setTimeout {
+    export function __promisify__(ms: number): Promise<void>;
```
Comment (rbuckton) [inline types/node/index.d.ts]: According to https://nodejs.org/api/timers.html#timers_settimeout_callback_delay_args, you can pass a value as well: `setTimeoutPromise(40, "foobar")`, so you might want to consider:
```ts
declare namespace setTimeout {
  export function __promisify__(ms: number): Promise<void>;
  export function __promisify__<T>(ms: number, value: T): Promise<T>;
}
```

File: types/node/index.d.ts [hunk 2]
```diff
@@ -81,10 +81,16 @@ declare var __filename: string;
...
+}
 declare function clearTimeout(timeoutId: NodeJS.Timer): void;
 declare function setInterval(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timer;
 declare function clearInterval(intervalId: NodeJS.Timer): void;
 declare function setImmediate(callback: (...args: any[]) => void, ...args: any[]): any;
+declare namespace setImmediate {
+    export function __promisify__(): Promise<void>;
```
Comment (rbuckton) [inline types/node/index.d.ts]: Similar to `setTimeout`, `setImmediate` can also take a value.

File: types/node/index.d.ts [hunk 3]
```diff
@@ -5666,10 +5672,16 @@ declare module "v8" {
 
 declare module "timers" {
     export function setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timer;
+    export namespace setTimeout {
+        export function __promisify__(ms: number): Promise<void>;
```
Comment (rbuckton) [inline types/node/index.d.ts]: See earlier comment.

File: types/node/index.d.ts [hunk 4]
```diff
@@ -5666,10 +5672,16 @@ declare module "v8" {
...
+    }
     export function clearTimeout(timeoutId: NodeJS.Timer): void;
     export function setInterval(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timer;
     export function clearInterval(intervalId: NodeJS.Timer): void;
     export function setImmediate(callback: (...args: any[]) => void, ...args: any[]): any;
+    export namespace setImmediate {
+        export function __promisify__(): Promise<void>;
```
Comment (rbuckton) [inline types/node/index.d.ts]: See earlier comment.

**Review by rbuckton — APPROVED** [review]
A few recommended additions, but this looks fine to me.

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
It looks like an unrelated change may have broken the CI build

**Pull request merged**